asllib/doc/Expressions.tex

```diff
@@ -1691,45 +1779,286 @@ \subsubsection{Formally}
+  \item All of the following apply (\textsc{okay}):
+  \begin{itemize}
+    \item applying $\assocopt$ to find the type associated with $\name$ in $\tfields$ yields the type $\vt$;
+    \item applying $\getbitvectorwidth$ to $\vt$ in $\tenv$ yields $\ewidth$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{error}):
+  \begin{itemize}
+    \item applying $\assocopt$ to find the type associated with $\name$ in $\tfields$ yields $\None$;
+    \item the result is a type error indicating that $\name$ is not associated with any bitfield in $\tfields$.
+  \end{itemize}
+\end{itemize}
+
+\subsubsection{Formally}
+\begin{mathpar}
+\inferrule[okay]{
+  \assocopt(\name, \tfields) \typearrow \langle\vt\rangle\\
+  \getbitvectorwidth(\tenv, \vt) \typearrow \ewidth
+}{
+  \getbitfieldwidth(\tenv, \name, \tfields) \typearrow \ewidth
+}
+\end{mathpar}
+
+\begin{mathpar}
+\inferrule[error]{
+  \assocopt(\name, \tfields) \typearrow \None
+}{
+  \getbitfieldwidth(\tenv, \name, \tfields) \typearrow \TypeErrorVal{BitFieldNotFound}
+}
+\end{mathpar}
+
+\subsubsection{TypingRule.WidthPlus}
+\hypertarget{def-widthplus}{}
+% NOTE: the implementation handles just two expressions, here we fold over a list.
+The helper function
+\[
+  \widthplus(\overname{\staticenvs}{\tenv} \aslsep \overname{\expr^*}{\exprs}) \typearrow \overname{\expr}{\ewidth}
+  \cup \overname{\TTypeError}{\TypeErrorConfig}
+\]
+generates the expression $\ewidth$, which represents the summation of all expressions in the list $\exprs$,
+normalized in the static environment $\tenv$.
+\ProseOrTypeError
+
+\subsubsection{Prose}
+One of the following applies:
+\begin{itemize}
+  \item All of the following apply (\textsc{empty}):
+  \begin{itemize}
+    \item $\exprs$ is an empty list;
+    \item $\ewidth$ is the literal expression for $0$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{non\_empty}):
+  \begin{itemize}
+    \item $\exprs$ is the list with \head\ $\ve$ and \tail\ $\exprsone$;
+    \item applying $\widthplus$ to $\exprsone$ yields $\ewidthone$;
+    \item applying $\normalize$ to the binary operation for $\PLUS$ and $\ve$ and $\ewidthone$ yields $\ewidth$\ProseOrTypeError.
+  \end{itemize}
+\end{itemize}
+
+\subsubsection{Formally}
+\begin{mathpar}
+\inferrule[empty]{}{
+  \widthplus(\tenv, \overname{\emptylist}{\exprs}) \typearrow \overname{\ELInt{0}}{\ewidth}
+}
+\end{mathpar}
+
+\begin{mathpar}
+\inferrule[non\_empty]{
+  \widthplus(\exprsone) \typearrow \ewidthone\\
+  \normalize(\tenv, \EBinop(\PLUS, \ve, \ewidthone)) \typearrow \ewidth \OrTypeError
+}{
+  \widthplus(\tenv, \overname{[\ve] \concat \exprsone}{\exprs}) \typearrow \ewidth
+}
+\end{mathpar}
+
+\subsection{Semantics}
+\subsubsection{SemanticsRule.EGetfields{\label{sec:SemanticsRule.EGetfields}}}
+\subsubsection{Prose}
+All of the following apply:
+\begin{itemize}
+  \item $\ve$ is the multi-field access expression for the expression $\erecord$ and list of field names
+        $\fieldnames$;
```

Comment:
typo?
```suggestion
        $v\fieldnames$;
```
Currently this shows as a blue/italicised function name (similarly below)